src/components/application_manager/src/help_prompt_manager_impl.cc

```diff
@@ -0,0 +1,436 @@
+                    "SetGlobalProperties RPC"
+                    " with the vrHelp and helpPrompt"
+                    " has already sent");
+      return;
+    }
+  }
+  StartTimer(is_restore);
+}
+
+void HelpPromptManagerImpl::OnAppUnregistered() {
+  LOG4CXX_AUTO_TRACE(logger_);
+  LOG4CXX_DEBUG(logger_, "Unregisted for appID:" << app_.app_id());
+  StopTimer();
+}
+
+void HelpPromptManagerImpl::StartTimer(bool is_restore) {
+  LOG4CXX_AUTO_TRACE(logger_);
+  LOG4CXX_DEBUG(logger_, "Timer for appID:" << app_.app_id());
+  StopTimer();
+  sending_type_ = SendingType::kSendBoth;
+  internal_timer_.Start(kBufferingTimeout, timer::kSingleShot);
+  if (is_restore) {
+    sending_type_ = is_tts_send_ ? SendingType::kSendVRHelp : sending_type_;
+    sending_type_ = is_ui_send_ ? SendingType::kSendHelpPromt : sending_type_;
+    is_tts_send_ = false;
+    is_ui_send_ = false;
+    const DataAccessor<CommandsMap> accessor = app_.commands_map();
+    const CommandsMap& commands = accessor.GetData();
+    for (auto& command : commands) {
+      OnVrCommandAdded(command.first, *command.second);
+    }
+  }
+}
+
+void HelpPromptManagerImpl::StopTimer() {
+  LOG4CXX_AUTO_TRACE(logger_);
+  sending_type_ = SendingType::kNoneSend;
+  if (internal_timer_.is_running()) {
+    internal_timer_.Stop();
+  }
+  vr_commands_.clear();
+}
+
+void HelpPromptManagerImpl::OnTimerExpired() {
+  LOG4CXX_AUTO_TRACE(logger_);
+  OnTimeoutExpired();
+}
+
+void HelpPromptManagerImpl::SendTTSRequest() {
+  LOG4CXX_AUTO_TRACE(logger_);
+  if (vr_commands_.empty()) {
+    LOG4CXX_DEBUG(logger_, "vr_commands_ is empty");
+    return;
+  }
+  LOG4CXX_DEBUG(logger_, "TTS request for appID:" << app_.app_id());
+  smart_objects::SmartObjectSPtr tts_global_properties =
+      utils::MakeShared<smart_objects::SmartObject>(
+          smart_objects::SmartType_Map);
+  if (tts_global_properties) {
+    smart_objects::SmartObject& ref = *tts_global_properties;
+
+    ref[strings::params][strings::message_type] =
+        hmi_apis::messageType::request;
+    ref[strings::params][strings::protocol_version] =
+        commands::CommandImpl::protocol_version_;
+    ref[strings::params][strings::protocol_type] =
+        commands::CommandImpl::hmi_protocol_type_;
+    ref[strings::params][strings::correlation_id] =
+        app_manager_.GetNextHMICorrelationID();
+
+    smart_objects::SmartObject& so_to_send = *tts_global_properties;
+    so_to_send[strings::params][strings::function_id] =
+        hmi_apis::FunctionID::TTS_SetGlobalProperties;
+
+    smart_objects::SmartObject msg_params =
+        smart_objects::SmartObject(smart_objects::SmartType_Map);
+
+    CreatePromtMsg(msg_params);
+
+    msg_params[strings::app_id] = app_.app_id();
+    so_to_send[strings::msg_params] = msg_params;
+    app_manager_.ManageHMICommand(tts_global_properties);
+  }
+}
+
+void HelpPromptManagerImpl::SendUIRequest() {
+  LOG4CXX_AUTO_TRACE(logger_);
+  if (vr_commands_.empty()) {
```

Comment:
@SNiukalov Are you sure that we have to check `vr_commands` in UI request?

Reply:
Yes, not send request if are not commands